internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -154,17 +155,29 @@ func (r *PreprovisioningImageReconciler) getIronicIgnitionConfig(log logrus.Fiel
 	return string(config), err
 }
 
-func (r *PreprovisioningImageReconciler) setImage(generation int64, status *metal3_v1alpha1.PreprovisioningImageStatus, infraEnv aiv1beta1.InfraEnv) error {
+func (r *PreprovisioningImageReconciler) setImage(image *metal3_v1alpha1.PreprovisioningImage, infraEnv aiv1beta1.InfraEnv) error {
 	r.Log.Infof("Updating PreprovisioningImage ImageUrl to: %s", infraEnv.Status.ISODownloadURL)
-	status.ImageUrl = infraEnv.Status.ISODownloadURL
-	status.Architecture = infraEnv.Spec.CpuArchitecture
-	status.Format = metal3_v1alpha1.ImageFormatISO
+	image.Status.Architecture = infraEnv.Spec.CpuArchitecture
+	if funk.Contains(image.Spec.AcceptFormats, metal3_v1alpha1.ImageFormatISO) {
+		r.Log.Infof("Updating PreprovisioningImage ImageUrl with ISO artifacts")
+		image.Status.Format = metal3_v1alpha1.ImageFormatISO
+		image.Status.ImageUrl = infraEnv.Status.ISODownloadURL
+		image.Status.KernelUrl = ""
+		image.Status.ExtraKernelParams = ""
+	} else if funk.Contains(image.Spec.AcceptFormats, metal3_v1alpha1.ImageFormatInitRD) {
+		r.Log.Infof("Updating PreprovisioningImage ImageUrl with InitRD artifacts")
+		image.Status.Format = metal3_v1alpha1.ImageFormatInitRD
+		image.Status.ImageUrl = infraEnv.Status.BootArtifacts.InitrdURL
+		image.Status.KernelUrl = infraEnv.Status.BootArtifacts.KernelURL
+		// should this respect existing settings? do some kind of search and append?
+		image.Status.ExtraKernelParams = fmt.Sprintf("coreos.live.rootfs_url=%s", infraEnv.Status.BootArtifacts.RootfsURL)
```

Comment:
The question in the comment is still something I'm curious about, but maybe it's just as easy as leaving that for a follow up.

The other question is if we need to include the rest of the kernel parameters here?
[All these, for example.](https://github.com/openshift/assisted-service/blob/6d3452d546b8a3c67068015e4e94ce64278792b5/internal/bminventory/inventory_v2_handlers.go#L619)

Reply:
I don't think we should include those, so that user would have an option to override some of these

Reply:
I suppose that's part of the question I asked originally in the comment.
Is this a value we expect users to be editing? If so, this logic needs to be significantly more sophisticated. If not, then we should probably be setting all those values, right?

Reply:
@eranco74 I think we talked about this and said "no we don't expect users to be touching this", but how else will they set kernel params on discovery?

cc @ori-amizur as you're working on that epic

Reply:
I guess there are few options for supporting this (annotations on the BMH, adding another attribute, getting it from infraEnv etc).
I think we can start with the current implementation and figure out how to support user provided kernel params later.